src/components/pseudonym/repo/pseudonym.repo.integration.test.js

```diff
@@ -30,34 +34,41 @@ describe('pseudonym repo', () => {
 	describe('deletePseudonyms', () => {
 		it('when pseudonym is deleted, it should return deleted pseudonym ids', async () => {
 			const user = await testObjects.createTestUser();
-			const ltiTool = await testObjects.createTestLtiTool();
-			const pseudonym = await testObjects.createTestPseudonym({ pseudonym: 'PSEUDONYM' }, ltiTool, user);
-			const result = await pseudonymRepo.deletePseudonyms([pseudonym._id]);
-			expect(result.length).to.be.equal(1);
-			expect(result[0]).to.be.equal(pseudonym._id);
+			await testObjects.createTestPseudonym({ pseudonym: 'PSEUDONYM' }, ltiTool, user);
+			const result = await pseudonymRepo.deletePseudonymsForUser(user._id);
+			expect(result.success).to.be.equal(true);
+			expect(result.deletedDocuments).to.be.equal(1);
 		});
 
 		it('when pseudonym is deleted, it should be gone from db', async () => {
 			const user = await testObjects.createTestUser();
-			const ltiTool = await testObjects.createTestLtiTool();
-			const pseudonym = await testObjects.createTestPseudonym({ pseudonym: 'PSEUDONYM' }, ltiTool, user);
-			await pseudonymRepo.deletePseudonyms([pseudonym._id]);
+			const pseudonym1 = await testObjects.createTestPseudonym({ pseudonym: 'PSEUDONYM' }, ltiTool, user);
+
+			const user2 = await testObjects.createTestUser();
+			const pseudonym2 = await testObjects.createTestPseudonym({ pseudonym: 'PSEUDONYM' }, ltiTool, user2);
+			let userPseudonyms = await pseudonymRepo.getPseudonymsForUser(user._id);
+			expect(userPseudonyms).to.be.an('array').of.length(1);
+			expect(userPseudonyms[0]._id.toString()).to.be.equal(pseudonym1._id.toString());
+
+			await pseudonymRepo.deletePseudonymsForUser(user._id);
 
-			const result = await Pseudonym.find({ userId: user._id }).lean().exec();
+			userPseudonyms = await pseudonymRepo.getPseudonymsForUser(user._id);
+			expect(userPseudonyms).to.be.an('array').of.length(0);
 
-			expect(result.length).to.be.equal(0);
+			const userPseudonyms2 = await pseudonymRepo.getPseudonymsForUser(user2._id);
+			expect(userPseudonyms2).to.be.an('array').of.length(1);
+			expect(userPseudonyms2[0]._id.toString()).to.be.equal(pseudonym2._id.toString());
 		});
 
 		it('when the function is called with invalid id, it throws an error', async () => {
 			const notExistedId = new ObjectId();
-			expect(pseudonymRepo.deletePseudonyms([notExistedId])).to.eventually.throw(new GeneralError());
+			expect(pseudonymRepo.deletePseudonymsForUser(notExistedId)).to.eventually.throw(GeneralError);
```

Comment:
should we ensure this or just return 0 documents removed instead as this returns always the counts and does not relate to a specific pseudonym document? doesn't this function just check the object-id format and not if the user exists?

Reply:
Yes the function checks the format of the userId and if the format is wrong it throws a GeneralError, because in this case it is unexpected behaviour and the execution should be stopped.